src/generator/generator.py

```diff
@@ -143,42 +144,44 @@ def generate_map_from_image(self, image_name: str, rand_entities: bool = False,
         return res_map
 
     def __in_bounds(self, pt: Point, dimensions: Size) -> bool:
-        return 0 <= pt.x < dimensions.width and 0 <= pt.y < dimensions.height
+        z = 0 <= pt.z < dimensions.depth if dimensions.n_dim == 3 else True
+        return 0 <= pt.x < dimensions.width and 0 <= pt.y < dimensions.height and z
 
     def __get_rand_position(self, dimensions: Size, start: Point = Point(0, 0)) -> Point:
```

Comment:
This is bad, should be using `start: Optional[Point]` and then setting start to a Point with correct dim if not given, otherwise you can't set start point for 3D.

Reply:
You may also want to merge `dimensions` and `start` into a `Union[Point, int]`, with dimensions being auto-determined given a point.